lib/core/network/models/transaction_model.dart

```diff
@@ -1,72 +1,336 @@
     required this.data,
     required this.actionName,
     required this.account,
     required this.timestamp,
-    required this.transactionId,
     required this.blockNumber,
+    required this.actor,
+    required this.transactionId,
   });
 
   @override
-  List<Object?> get props => [transactionId];
+  List<Object?> get props => [actionName, data, account, timestamp, blockNumber];
 
   factory TransactionModel.fromJson(Map<String, dynamic> json) {
-    return TransactionModel(
-      account: json['act']['account'],
-      actionName: json['act']['name'],
-      data: json['act']['data'],
-      timestamp: parseTimestamp(json['@timestamp']),
-      transactionId: json['trx_id'],
-      blockNumber: json['block_num'],
-    );
-  }
+    final act = json['act'];
+    final actionName = act['name'];
+    final account = act['account'];
+    final data = act['data'];
+    final timestamp = parseTimestamp(json['@timestamp']);
+    final blockNumber = json['block_num'];
+    final actor = act['authorization'].first['actor'];
+    final transactionId = json['trx_id'];
 
-// factory TransactionModel.fromJsonMongo(Map<String, dynamic> json) {
-//   return TransactionModel(
-//     account: json['act']['account'],
-//     actionName: json['act']['name'],
-//     data: json['act']['data'],
-//     timestamp: parseTimestamp(json['@timestamp']),
-//     transactionId: json['trx_id'],
-//     blockNumber: json['  block_num'],
-//   );
-// }
-
-// static TransactionModel? fromTransaction(GenericTransactionModel genericModel) {
-//   if (genericModel.transaction.isTransfer) {
-//     final action = genericModel.transaction.actions.first;
-//     final data = action.data;
-//     final String? from = data?['from'];
-//     final String? to = data?['to'];
-//     final String? quantity = data?['quantity'];
-//     final String memo = data?['memo'] ?? "";
-//     if (from != null && to != null && quantity != null) {
-//       return TransactionModel(
-//         from: from,
-//         to: to,
-//         quantity: quantity,
-//         memo: memo,
-//         timestamp: genericModel.timestamp ?? DateTime.now().toUtc(),
-//         transactionId: genericModel.transactionId,
-//       );
-//     }
-//   }
-//
-//   return null;
-// }
+    switch (actionName) {
+      case 'transfer':
+        {
+          return TransactionTransfer(
+            account: account,
+            actionName: actionName,
+            blockNumber: blockNumber,
+            data: data,
+            timestamp: timestamp,
+            actor: actor,
+            transactionId: transactionId,
+          );
+        }
+      case 'redeem':
+        {
+          return TransactionRedeem(
+            account: account,
+            actionName: actionName,
+            blockNumber: blockNumber,
+            data: data,
+            timestamp: timestamp,
+            actor: actor,
+            transactionId: transactionId,
+          );
+        }
+      case 'cmntadd':
+        {
```

Comment:
same for all these... need to check which contract the action is run on and it needs to be dao.hypha